sound/soc/sof/topology.c

```diff
@@ -338,11 +338,17 @@ static int sof_control_load_bytes(struct snd_soc_component *scomp,
 	struct sof_ipc_ctrl_data *cdata;
 	struct snd_soc_tplg_bytes_control *control =
 		(struct snd_soc_tplg_bytes_control *)hdr;
-	const int max_size = SOF_IPC_MSG_MAX_SIZE -
-		sizeof(const struct sof_ipc_ctrl_data);
+	struct soc_bytes_ext *sbe = (struct soc_bytes_ext *)kc->private_value;
+	int max_size = sbe->max;
+
+	if (le32_to_cpu(control->priv.size) > max_size) {
+		dev_err(sdev->dev, "err: bytes data size %d exceeds max %d.\n",
+			control->priv.size, max_size);
+		return -EINVAL;
+	}
 
 	/* init the get/put bytes data */
-	scontrol->size = SOF_IPC_MSG_MAX_SIZE;
+	scontrol->size = control->priv.size;
```

Comment:
should this also be le32_to_cpu()?

Reply:
yes, you want to check the new sparse warnings:
sound/soc/sof/topology.c:351:24: warning: incorrect type in assignment (different base types)

Reply:
how/where should I navigate to see the sparse warnings... the build sparse log looked clean...?

Reply:
@juimonen click on jenkins checkpatch below to see the warnings